modules/ximgproc/src/sparse_match_interpolators.cpp

```diff
@@ -877,5 +887,1026 @@ bool operator<(const SparseMatch& lhs,const SparseMatch& rhs)
+        Mat & matNNDis, vector<int> & outSupportIds, vector<float> &  outSupportDis);
+    float GetWeightFromDistance(float dis);
+    int PropagateModels(int spCnt, Mat & spNN, vector<int> & supportMatchIds, vector<float> & supportMatchDis, int supportCnt,
+        const vector<SparseMatch> &inputMatches, Mat & outModels);
+    float HypothesisEvaluation(const Mat & inModel, const int * matNodes, const float * matDis, int matCnt, const vector<SparseMatch> & inputMatches, Mat & outInLier);
+    int HypothesisGeneration(int* matNodes, int matCnt, const vector<SparseMatch> & inputMatches, Mat & outModel);
+
+public:
+    void setCostMap(const Mat & _costMap) CV_OVERRIDE { _costMap.copyTo(costMap); }
+    void setK(int val) CV_OVERRIDE { max_neighbors  = val; }
+    int  getK() const CV_OVERRIDE { return max_neighbors; }
+    void setSuperpixelSize(int val) CV_OVERRIDE { sp_size = val; }
+    int  getSuperpixelSize() const CV_OVERRIDE { return sp_size; }
+    void setSuperpixelNNCnt(int val) CV_OVERRIDE { sp_nncnt = val; }
+    int  getSuperpixelNNCnt() const CV_OVERRIDE { return sp_nncnt; }
+    void setSuperpixelRuler(float val) CV_OVERRIDE { sp_ruler = val; }
+    float getSuperpixelRuler() const CV_OVERRIDE { return sp_ruler; }
+    void setSuperpixelMode(int val) CV_OVERRIDE
+    {
+        slic_type = static_cast<SLICType>(val);
+        CV_Assert(slic_type == SLICO || slic_type == SLIC || slic_type == MSLIC);
+    }
+    int  getSuperpixelMode() const CV_OVERRIDE { return slic_type; }
+    void setAlpha(float val) CV_OVERRIDE { alpha = val; }
+    float getAlpha() const CV_OVERRIDE { return alpha; }
+    void setModelIter(int val) CV_OVERRIDE { model_iter = val; }
+    int  getModelIter() const CV_OVERRIDE { return model_iter; }
+    void setRefineModels(bool val) CV_OVERRIDE { refine_models = static_cast<int>(val); }
+    bool getRefineModels() const CV_OVERRIDE { return refine_models; }
+    void setMaxFlow(float val) CV_OVERRIDE { max_flow = val; }
+    float getMaxFlow() const CV_OVERRIDE { return max_flow; }
+    void setUseVariationalRefinement(bool val) CV_OVERRIDE { use_variational_refinement = val; }
+    bool getUseVariationalRefinement() const CV_OVERRIDE { return use_variational_refinement; }
+    void  setUseGlobalSmootherFilter(bool val) CV_OVERRIDE { use_global_smoother_filter = val; }
+    bool  getUseGlobalSmootherFilter() const CV_OVERRIDE { return use_global_smoother_filter; }
+    void  setFGSLambda(float _lambda) CV_OVERRIDE { fgs_lambda = _lambda; }
+    float getFGSLambda() const CV_OVERRIDE { return fgs_lambda; }
+    void  setFGSSigma(float _sigma) CV_OVERRIDE { fgs_sigma = _sigma; }
+    float getFGSSigma() const CV_OVERRIDE { return fgs_sigma; }
+};
+
+Ptr<RICInterpolatorImpl> RICInterpolatorImpl::create()
+{
+    RICInterpolatorImpl *eai = new RICInterpolatorImpl();
+    eai->init();
+    return Ptr<RICInterpolatorImpl>(eai);
```

Comment:
Memory leak is possible in case of exception in "init".
Don't use raw pointers:
```
auto eai = makePtr<RICInterpolatorImpl>();
eai->init()
return eai;
```

Reply:
Thank you for that hint!